comps/dataprep/pgvector/prepare_doc_pgvector.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import os
+import uuid
+from pathlib import Path
+from typing import List, Optional, Union
+
+from config import EMBED_MODEL, INDEX_NAME, PG_CONNECTION_STRING
+from fastapi import File, Form, HTTPException, UploadFile
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceHubEmbeddings
+from langchain_community.vectorstores import PGVector
+from langsmith import traceable
+
+from comps import DocPath, opea_microservices, register_microservice
+from comps.dataprep.utils import document_loader, parse_html
+
+tei_embedding_endpoint = os.getenv("TEI_ENDPOINT")
+
+
+async def save_file_to_local_disk(save_path: str, file):
+    save_path = Path(save_path)
+    with save_path.open("wb") as fout:
+        try:
+            content = await file.read()
+            fout.write(content)
+        except Exception as e:
+            print(f"Write file failed. Exception: {e}")
+            raise HTTPException(status_code=500, detail=f"Write file {save_path} failed. Exception: {e}")
+
+
+def ingest_data_to_pgvector(doc_path: DocPath):
```

Comment:
ingest_doc_to_pgvector instead of current name?

Reply:
Done